pkg/podmounter/mountoptions/mount_options.go

```diff
@@ -154,12 +156,59 @@ func parseUnixRights(buf []byte) ([]int, error) {
 	return fds, nil
 }
 
+// tryToMakeSockPathRelative tries to make `path` relative to the current working directory
+// if its longer than 108 characters. This is because Go returns `invalid argument` errors
+// if you try to do `net.Listen()` or `net.Dial()`, see https://github.com/golang/go/issues/6895 for more details.
+//
+// If the `path` is still longer than 108 characters after making it relative, this function emits a warning log.
+// If the `path` is not relative, this function just returns unmodified `path` and emits a warning log if needed.
+func tryToMakeSockPathRelative(path string) string {
+	if len(path) < 108 {
+		// Nothing to do, its already shorter than 108 characters
+		return path
+	}
+
+	shortPath, err := relative(path)
```

Comment:
We should warn/info anyway here in case something goes wrong

Reply:
Not sure if this error log would be useful/actionable by the user though. We just try to make `path` relative to `cwd` and it might fail in some conditions, but there isn't much can be done by the user in those cases, as `cwd` is determined by our process.

We'd log if the Unix socket is longer still.

Reply:
Added a warning with https://github.com/awslabs/mountpoint-s3-csi-driver/pull/368/commits/cb539cc9e7eae43267799d38cebd3c85f2a5d13f, we can remove it if it gets verbose/noisy